api/src/main/java/org/eclipse/microprofile/lra/annotation/Forget.java

```diff
@@ -28,46 +28,91 @@
+ * </p>
  *
+ * <p>
  * A similar remark applies if the participant was enlisted in a
- * nested LRA {@link LRA#value()}. Actions performed in the context
+ * nested LRA {@link LRA.Type#NESTED}. Actions performed in the context
  * of a nested LRA must remain compensatable until the participant
- * is explicitly told it can clean up using this <em>@Forget</em>
+ * is explicitly told it can clean up using this <code>&#64;Forget</code>
  * annotation.
+ * </p>
  *
+ * <p>
  * If the annotated method is a JAX-RS resource method the id of the currently
  * running LRA can be obtained by inspecting the incoming JAX-RS headers. If
  * this LRA is nested then the parent LRA MUST be present in the header with the name
- * {@link org.eclipse.microprofile.lra.annotation.ws.rs.LRA#LRA_HTTP_PARENT_CONTEXT_HEADER}
+ * {@link LRA#LRA_HTTP_PARENT_CONTEXT_HEADER}
  * and value is of type {@link java.net.URI}.
+ * </p>
  *
+ * <p>
  * If the annotated method is not a JAX-RS resource method the id of the currently
  * running LRA can be obtained by adhering to a predefined method signature as
  * defined in the LRA specification document. Similarly the method may determine
  * whether or not it runs with a nested LRA by providing a parameter to hold the parent id.
  * For example,
+ * </p>
+ *
  * <pre>
  *     <code>
  *          &#64;Forget
  *          public void forget(URI lraId, URI parentId) { ...}
  *     </code>
  * </pre>
+ *
+ * <p>
  * would be a valid forget method declaration. If an invalid signature is detected 
  * the implementation of this specification MUST prohibit successful startup of the application
  * (e.g. with a runtime exception).
+ * </p>
  *
- *
+ * <p>
  * Since the participant generally needs to know the id of the LRA in order
  * to clean up there is generally no benefit to combining this annotation
- * with the `@LRA` annotation (though it is not prohibited).
+ * with the <code>&#64;LRA</code> annotation (though it is not prohibited).
+ * </p>
  *
+ * <p>
  * Related information is provided in the javadoc for the {@link Status}
+ * </p>
+ * <p>
+ * If the method is a JAX-RS resource method (or is a non JAX-RS method
+ * annotated with <code>&#64;Forget</code> with return type
+ * <code>javax.ws.rs.core.Response)</code> then the following are the only
+ * valid response codes:
+ * </p>
+ *
+ *<table border="0" cellpadding="3" cellspacing="0"
+ *   summary="Valid JAX-RS response codes for Forget methods">
+ * <caption><span>JAX-RS Response Codes For Forget Methods</span><span>&nbsp;</span></caption>
+ * <tr>
+ *   <th scope="col">Code</th>
+ *   <th scope="col">Response Body</th>
+ *   <th scope="col">Meaning</th>
+ * </tr>
+ * <tr>
+ *   <td scope="row">200</td>
+ *   <td scope="row">Empty</td>
+ *   <td scope="row">The participant may have removed all knowledge of the LRA</td>
+ * </tr>
+ * <tr>
+ *   <td scope="row">404</td>
+ *   <td scope="row">Empty</td>
+ *   <td scope="row">The method does not know about the LRA</td>
```

Comment:
does that distinction 200 vs 404 have any meaning for the caller? This is not important, just unnecessary for the developer to think about the returned HTTP response code when it's not being used.

Reply:
No but it has advantages. All the other methods use 404 and one could imagine a generic precondition check `if  LRA does not exist then throw WebApplicationException with a 404 response code`. Without it the developer needs to think I don't know about this LRA what response should I return so he still needs to think about it.